allure-app-under-test/src/main/java/com/capgemini/ntc/selenium/pages/projectY/HorizontalSliderPage.java

```diff
@@ -0,0 +1,231 @@
+	
+	private static final int	MOUSE_MOVE_LEFT		= 0;
+	private static final int	MOUSE_MOVE_RIGHT	= 1;
+	
+	private static final By	selectorHorizontalSlider	= By.cssSelector("div.sliderContainer");
+	private static final By	sliderSelector				= By.cssSelector("input");
+	private static final By	valueSelector				= By.cssSelector("#range");
+	
+	private HorizontalSliderElement horizontalSlider;
+	
+	public HorizontalSliderPage() {
+		horizontalSlider = getDriver().elementHorizontalSlider(selectorHorizontalSlider, sliderSelector, valueSelector, BigDecimal.ZERO, new BigDecimal(5), new BigDecimal(0.5));
+	}
+	
+	@Override
+	public boolean isLoaded() {
+		getDriver().waitForPageLoaded();
+		return getDriver().getCurrentUrl()
+						.contains("horizontal_slider");
+	}
+	
+	@Override
+	public void load() {
+		BFLogger.logDebug("load()");
+		
+	}
+	
+	@Override
+	public String pageTitle() {
+		return "The Internet";
+	}
+	
+	public boolean isElementHorizontalSliderVisible() {
+		return getDriver().elementHorizontalSlider(selectorHorizontalSlider)
+						.isDisplayed();
+	}
+	
+	/**
+	 * Returns the value of slider's start position.
+	 * 
+	 * @return BigDecimal representing the lowest possible value of slider.
+	 */
+	public BigDecimal getStartPosition() {
+		return horizontalSlider.getMinRange();
+	}
+	
+	/**
+	 * Returns the value of slider's middle position.
+	 * 
+	 * @return BigDecimal representing the average value between start and end position.
+	 */
+	public BigDecimal getMiddlePosition() {
+		return horizontalSlider.getMaxRange()
+						.subtract(horizontalSlider.getMinRange())
+						.divide(new BigDecimal(2));
+	}
+	
+	/**
+	 * Returns the value of slider's end position.
+	 * 
+	 * @return BigDecimal representing the highest possible value of slider.
+	 */
+	public BigDecimal getEndPosition() {
+		return horizontalSlider.getMaxRange();
+	}
+	
+	/**
+	 * Returns current value of slider's position.
+	 * 
+	 * @return BigDecimal representing current value of slider.
+	 */
+	public BigDecimal getCurrentPosition() {
+		return horizontalSlider.getCurrentSliderValue();
+	}
+	
+	/**
+	 * Sets slider to the position given by user using one of following methods:
+	 * keyboard button press or mouse move.
+	 * <p>
+	 * If position chosen by user is less than the minimum or more than the maximum possible value,
+	 * then position is automatically limited to minimum or maximum value.
+	 * </p>
+	 * <p>
+	 * Also, if position chosen by user has a value which is not a multiple of step defined for
+	 * horizontal slider, then it is rounded to nearest proper value.
+	 * </p>
+	 * 
+	 * @param position
+	 *            A destination value of slider's position.
+	 * @param method
+	 *            The way to perform slider's movement:
+	 *            <ul>
+	 *            <li>0 - KEYBOARD
+	 *            <li>1 - MOUSE
+	 *            </ul>
+	 * @see HorizontalSliderElement
+	 */
+	public void setSliderPositionTo(BigDecimal position, int method) {
```

Comment:
Is it possible to move this to "HorizontalSliderElement.class"  ? Could be it be a generic  set slider position ?

Reply:
Method has been moved. Also left mouse click has been added as suggested in PR #75 to web drivers.